src/main/scala/it/pagopa/interop/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -121,6 +121,88 @@ class PartyApiServiceImpl(
 
   }
 
+  /** Code: 200, Message: successful operation, DataType: Institution
+   * Code: 405, Message: Institution not found, DataType: Problem
+   * Code: 400, Message: Invalid ID supplied, DataType: Problem
+   */
+  override def updateInstitutionById(id: String, institution: Institution)(implicit
+    toEntityMarshallerInstitution: ToEntityMarshaller[Institution],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Updating institution $id")
+
+    val commander: EntityRef[Command] = getCommander(id)
+
+    val updatedInstitution: Future[StatusReply[Party]] = for {
+      uuid       <- id.toFutureUUID
+      found      <- commander.ask(ref => GetParty(uuid, ref))
+      party      <- found.toFuture(UpdateInstitutionNotFound(id)).map(InstitutionParty.extractInstitutionParty)
+      updatedOrg <- party
+        .map(p => {
+          if (p.externalId != institution.institutionId) {
+            Left(
+              Failure(
+                UpdateInstitutionBadRequest(
+                  id,
+                  s"Cannot update externalId ${p.externalId} -> ${institution.institutionId}"
+                )
+              )
+            )
+          } else {
+            Right(
+              InstitutionParty(
+                id = uuid,
+                externalId = institution.institutionId,
+                description = institution.description,
+                digitalAddress = institution.digitalAddress,
+                taxCode = institution.taxCode,
+                address = institution.address,
+                zipCode = institution.zipCode,
+                origin = institution.origin,
+                institutionType = institution.institutionType,
+                start = p.start,
+                end = p.end,
+                attributes = institution.attributes.map(InstitutionAttribute.fromApi).toSet
+              )
+            )
+          }
+        })
+      result     <- updatedOrg
+        .fold(e => e.toFuture, p => commander.ask(ref => UpdateParty(p, ref)))
+    } yield result
```

Comment:
I'm a bit confused by the errorHandling here. Why is there a Try monad (i.e. Failure) wrapped in Either monad (i.e. Left)? Plus it's everything inside a Future too. If you are okay with that I'll commit a suggestion to this error handling.

Reply:
may I suggest to use a suggestion block here so that your advices are easily readable?

Reply:
I pushed a commit after having it reviewed by @antonioT90 together in a call